include/ByteBuffer.hpp

```diff
@@ -407,6 +407,14 @@ class ByteBuffer {
     int32_t capacity() {
         return m_capacity;
     }
+
+    bool operator!=(const ByteBuffer& other) {
+        bool isEqual = (m_capacity == other.m_capacity);
+        isEqual = (isEqual && (memcmp(m_buffer, other.m_buffer, m_capacity) == 0));
```

Comment:
This seems to be too strict. Might want to consider comparing `m_limit` and then limit the `memcmp` to `m_limit` in case one of the `m_buffer` has garbage after `m_limit` bytes.